IPython/core/display.py

```diff
@@ -666,6 +666,11 @@ def _repr_pretty_(self, pp, cycle):
 
 class HTML(TextDisplayObject):
 
+    def __init__(self, data=None, url=None, filename=None, metadata=None):
+        if data and "<iframe " in data and "</iframe>" in data:
```

Comment:
Wondering if we should use `startswith` and `endswith` that are more efficient and likely more correct, but that looks good.

Reply:
Yeah, I had the same thought. There’s a difference between rendering a single iframe vs rendering HTML with an iframe in it. Startswith would be a better idea. We could also account for case differences because HTML is case insensitive.

Let me know if you’d like those changes, or if things are fine as they are.

Reply:
Up to you. I'll reopen the linked issue add link to this section for further improvements.